### DISCUSSION TITLE
Shift to Compiled Language for High Performance and Low Latency

### DISCUSSION THREAD

**Discussion by sirinath** [geldata/gel#1309] [Ideas]
I noticed a large portion of the codebase is written in  Python. Is it possible to gradually move to compiled language? E.g.  [Nim](https://nim-lang.org/) has a Python-like syntax but generates C/C++ which can be compiled. Also [Pythran](https://github.com/serge-sans-paille/pythran) and [Cython](https://cython.org/) might do the trick.

**Comment by elprans** [comment]
EdgeDB is already written in a mix of Python, Cython, and Rust. The proportion of compiled code will likely increase over time, but even now the overhead of EdgeDB over Postgres when running prepared queries is quite low: [benchmark](https://edgedb.com/blog/edgedb-1-0-alpha-2/#results).

**Comment by sirinath** [comment]
Great!

Also there is [Nuitka ](https://nuitka.net/) to speed up Python. Performance is never enough so whatever opportunity to fine-tune please take it.
